app/src/ui/repository.tsx

```diff
@@ -283,15 +285,31 @@ export class RepositoryView extends React.Component<
           return null
         }
 
-        return (
-          <div>
-            <p>
-              {`${stashEntry.name} created on ${
-                stashEntry.branchName
-              }@${stashEntry.stashSha.substr(0, 7)}`}
-            </p>
-          </div>
-        )
+        if (Array.isArray(stashEntry.files)) {
```

Comment:
Changing the type definition for `StashedFileChanges` will remove the need for this check.